Services/Init/classes/class.ilStartUpGUI.php

```diff
@@ -1296,6 +1316,17 @@ public function doLogout()
 
         $user_language = $user->getLanguage();
 
+
+        if(isset($_GET['withdraw_consent'])) {
+            //handling withdrawal of consent request to be able to catch it on relog
+            $user->writePref('consent_withdrawal_requested', 1);
```

Comment:
Please factor this out to a new `TermsOfService` class. The footprint in `ilStartupGUI` should be as small as possible. Especially because you are changing state here ... Maybe you could throw an event here and change the state in a corresponding listener.